tests/ftqc/test_pauli_tracker.py

```diff
@@ -83,3 +83,102 @@ def test_pauli_prod(self, ops, expected):
+            (qml.S(1), [qml.X(1)], [qml.Y(1)]),
+            (qml.S(2), [qml.Y(2)], [qml.X(2)]),
+            (qml.S(3), [qml.Z(3)], [qml.Z(3)]),
+            (qml.H(0), [qml.I(0)], [qml.I(0)]),
+            (qml.H(1), [qml.X(1)], [qml.Z(1)]),
+            (qml.H(2), [qml.Y(2)], [qml.Y(2)]),
+            (qml.H(3), [qml.Z(3)], [qml.X(3)]),
+            (qml.CNOT(wires=[0, 1]), [qml.I(0), qml.I(1)], [qml.I(0), qml.I(1)]),
+            (qml.CNOT(wires=[0, 1]), [qml.X(0), qml.I(1)], [qml.X(0), qml.X(1)]),
+            (qml.CNOT(wires=[0, 1]), [qml.Y(0), qml.I(1)], [qml.Y(0), qml.X(1)]),
+            (qml.CNOT(wires=[0, 1]), [qml.Z(0), qml.I(1)], [qml.Z(0), qml.I(1)]),
+            (qml.CNOT(wires=[0, 1]), [qml.I(0), qml.X(1)], [qml.I(0), qml.X(1)]),
+            (qml.CNOT(wires=[0, 1]), [qml.X(0), qml.X(1)], [qml.X(0), qml.I(1)]),
+            (qml.CNOT(wires=[0, 1]), [qml.Y(0), qml.X(1)], [qml.Y(0), qml.I(1)]),
+            (qml.CNOT(wires=[0, 1]), [qml.Z(0), qml.X(1)], [qml.Z(0), qml.X(1)]),
+            (qml.CNOT(wires=[0, 1]), [qml.I(0), qml.Y(1)], [qml.Z(0), qml.Y(1)]),
+            (qml.CNOT(wires=[0, 1]), [qml.X(0), qml.Y(1)], [qml.Y(0), qml.Z(1)]),
+            (qml.CNOT(wires=[0, 2]), [qml.Y(0), qml.Y(2)], [qml.X(0), qml.Z(2)]),
+            (qml.CNOT(wires=[0, 2]), [qml.Z(0), qml.Y(2)], [qml.I(0), qml.Y(2)]),
+            (qml.CNOT(wires=[1, 2]), [qml.I(1), qml.Z(2)], [qml.Z(1), qml.Z(2)]),
+            (qml.CNOT(wires=[1, 2]), [qml.X(1), qml.Z(2)], [qml.Y(1), qml.Y(2)]),
+            (qml.CNOT(wires=[1, 2]), [qml.Y(1), qml.Z(2)], [qml.X(1), qml.Y(2)]),
+            (qml.CNOT(wires=[1, 2]), [qml.Z(1), qml.Z(2)], [qml.I(1), qml.Z(2)]),
+            (qml.CNOT(wires=[0, 1]), [qml.I(1), qml.I(0)], [qml.I(0), qml.I(1)]),
+            (qml.CNOT(wires=[0, 1]), [qml.X(1), qml.I(0)], [qml.I(0), qml.X(1)]),
+            (qml.CNOT(wires=[0, 1]), [qml.Y(1), qml.I(0)], [qml.Z(0), qml.Y(1)]),
+            (qml.CNOT(wires=[0, 1]), [qml.Z(1), qml.I(0)], [qml.Z(0), qml.Z(1)]),
+            (qml.CNOT(wires=[0, 1]), [qml.I(1), qml.X(0)], [qml.X(0), qml.X(1)]),
+            (qml.CNOT(wires=[0, 1]), [qml.X(1), qml.X(0)], [qml.X(0), qml.I(1)]),
+            (qml.CNOT(wires=[0, 1]), [qml.Y(1), qml.X(0)], [qml.Y(0), qml.Z(1)]),
+            (qml.CNOT(wires=[0, 1]), [qml.Z(1), qml.X(0)], [qml.Y(0), qml.Y(1)]),
+            (qml.CNOT(wires=[0, 1]), [qml.I(1), qml.Y(0)], [qml.Y(0), qml.X(1)]),
+            (qml.CNOT(wires=[0, 1]), [qml.X(1), qml.Y(0)], [qml.Y(0), qml.I(1)]),
+            (qml.CNOT(wires=[0, 2]), [qml.Y(2), qml.Y(0)], [qml.X(0), qml.Z(2)]),
+            (qml.CNOT(wires=[0, 2]), [qml.Z(2), qml.Y(0)], [qml.X(0), qml.Y(2)]),
+            (qml.CNOT(wires=[1, 2]), [qml.I(2), qml.Z(1)], [qml.Z(1), qml.I(2)]),
+            (qml.CNOT(wires=[1, 2]), [qml.X(2), qml.Z(1)], [qml.Z(1), qml.X(2)]),
+            (qml.CNOT(wires=[1, 2]), [qml.Y(2), qml.Z(1)], [qml.I(1), qml.Y(2)]),
+            (qml.CNOT(wires=[1, 2]), [qml.Z(2), qml.Z(1)], [qml.I(1), qml.Z(2)]),
```

Comment:
Can we swap the order? The input for `CNOT` is ordered `(ctrl, target)`, so lets make the order for the list input when working with CNOT also `ctrl, target`

Reply:
The tests for the input Pauli target wire matches (ctrl, target) are L98-L113. The tests L114-L129 are for out of order scenarios. We can remove them if needed.

Reply:
Ah, I missed that, thanks!